tests/test_pretrial_verified_law_rescue.py

```
@@ -0,0 +1,124 @@
+    assert "текст нормы" not in basis[0]
+    assert "источник:" not in basis[0]
+
+
+def test_pretrial_repairs_bad_article_from_verified_research_without_second_llm_call() -> None:
+    async def scenario() -> None:
+        research = _verified_research()
+        initial = _draft_with_bad_article()
+        assert pretrial_quality_issues(initial, research)
+        calls: list[dict] = []
+
+        async def original(_self, _context, _research, language="ru"):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `original`

(ANN202)